benchmark/alpa/benchmark_2d_one_case_moe.py

```diff
@@ -61,21 +59,23 @@ def benchmark_moe_internal(physical_mesh, benchmark_case, niter):
         num_micro_batches = None
         grad_func = jax.grad
 
+    as_option = {}
     if force_batch_dim_mapping:
         # Always map batch dim to mesh dim 0
-        as_option.force_batch_dim_to_mesh_dim = 0
-    as_option.prefer_reduce_scatter = prefer_reduce_scatter
-    as_option.allow_mixed_mesh_shape = True
+        as_option["force_batch_dim_to_mesh_dim"] = 0
+    as_option["prefer_reduce_scatter"] = prefer_reduce_scatter
+    as_option["allow_mixed_mesh_shape"] = True
 
     if parallel_mode == "zero-3":
-        as_option.force_zero_stage_3 = True
+        as_option["force_zero_stage_3"] = True
     elif parallel_mode in ["shard-largest"]:
-        as_option.force_simple_heuristic = other
+        as_option["force_simple_heuristic"] = other
```

Comment:
Can we create an `as_option = AutoShardingOption()` and then modify the fields of the `as_option` by `as_option.xxx = yyy` instead of creating a dictionary? The latter option can prevent setting the wrong field.